backend/internal/api/handlers.go

```diff
@@ -0,0 +1,299 @@
+// Package api provides HTTP handlers with repository access
+package api
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/ali01/mnemosyne/internal/config"
+	"github.com/ali01/mnemosyne/internal/models"
+	"github.com/ali01/mnemosyne/internal/repository"
+	"github.com/gin-gonic/gin"
+)
+
+// Handler holds dependencies for HTTP handlers
+type Handler struct {
+	repos *repository.Repositories
+	cfg   *config.Config
+}
+
+// getGraph returns the full graph data from the database
+func (h *Handler) getGraph(c *gin.Context) {
+	ctx := context.Background()
+
+	// Get pagination parameters
+	limit, _ := strconv.Atoi(c.DefaultQuery("limit", "1000"))
+	offset, _ := strconv.Atoi(c.DefaultQuery("offset", "0"))
+
+	// Get all nodes
+	vaultNodes, err := h.repos.Nodes.GetAll(ctx, limit, offset)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to fetch nodes"})
+		return
+	}
+
+	// Get all edges
+	vaultEdges, err := h.repos.Edges.GetAll(ctx, limit, offset)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to fetch edges"})
+		return
+	}
+
+	// Get all positions
+	positions, err := h.repos.Positions.GetAll(ctx)
+	if err != nil {
+		// Non-fatal: positions are optional
+		positions = []repository.NodePosition{}
```

Comment:
Good defensive programming here. Treating position fetch errors as non-fatal ensures the API remains functional even if the positions table has issues. This graceful degradation is important for maintaining service availability.